pkg/envoy/envoy.go

```diff
@@ -242,6 +247,68 @@ func StartEnvoy(adminPort uint32, stateDir, logPath string, baseID uint64) *Envo
 	return nil
 }
 
+// newEnvoyLogPiper creates a writer that parses and logs log messages written by Envoy.
+func newEnvoyLogPiper() io.WriteCloser {
+	reader, writer := io.Pipe()
+	scanner := bufio.NewScanner(reader)
+	go func() {
+		scopedLog := log.WithFields(logrus.Fields{
+			logfields.LogSubsys: "unknown",
+			logfields.ThreadID:  "unknown",
+		})
+		level := "debug"
+
+		for scanner.Scan() {
+			line := scanner.Text()
+			var msg string
+
+			parts := strings.SplitN(line, "|", 4)
+			// Parse the line as a log message written by Envoy, assuming it
+			// uses the configured format: "%t|%l|%n|%v".
+			if len(parts) == 4 {
+				threadID := parts[0]
+				level = parts[1]
+				loggerName := parts[2]
+				// TODO: Parse msg to extract the source filename, line number, etc.
+				msg = fmt.Sprintf("[%s", parts[3])
```

Comment:
Shouldn't this be `[%s]`?

Reply:
No, because the Envoy code is quite tied to the default format, the messages actually look like this:
```
external/envoy/source/server/server.cc:347] caught SIGTERM
```
i.e. it expects the log format to prepend the message with `[`. I'm doing just that.